Kernel/Firmware/EFI/EFI.h

```diff
@@ -0,0 +1,211 @@
+// Standard status codes are defined here: https://uefi.org/specs/UEFI/2.10/Apx_D_Status_Codes.html
+static constexpr FlatPtr ERROR_MASK = 1ull << (NumericLimits<FlatPtr>::digits() - 1);
+enum class Status : FlatPtr {
+    Success = 0,
+
+    LoadError = 1 | ERROR_MASK,
+    InvalidParameter = 2 | ERROR_MASK,
+    Unsupported = 3 | ERROR_MASK,
+    BadBufferSize = 4 | ERROR_MASK,
+    BufferTooSmall = 5 | ERROR_MASK,
+    NotReady = 6 | ERROR_MASK,
+    DeviceError = 7 | ERROR_MASK,
+    WriteProtected = 8 | ERROR_MASK,
+    OutOfResources = 9 | ERROR_MASK,
+    VolumeCorrupted = 10 | ERROR_MASK,
+    VolumeFull = 11 | ERROR_MASK,
+    NoMedia = 12 | ERROR_MASK,
+    MediaChanged = 13 | ERROR_MASK,
+    NotFound = 14 | ERROR_MASK,
+    AccessDenied = 15 | ERROR_MASK,
+    NoResponse = 16 | ERROR_MASK,
+    NoMapping = 17 | ERROR_MASK,
+    Timeout = 18 | ERROR_MASK,
+    NotStarted = 19 | ERROR_MASK,
+    AlreadyStarted = 20 | ERROR_MASK,
+    Aborted = 21 | ERROR_MASK,
+    ICMPError = 22 | ERROR_MASK,
+    TFTPError = 23 | ERROR_MASK,
+    ProtocolError = 24 | ERROR_MASK,
+    IncompatibleVersion = 25 | ERROR_MASK,
+    SecurityViolation = 26 | ERROR_MASK,
+    CRCError = 27 | ERROR_MASK,
+    EndOfMedia = 28 | ERROR_MASK,
+
+    EndOfFile = 31 | ERROR_MASK,
+    InvalidLanguage = 32 | ERROR_MASK,
+    CompromisedData = 33 | ERROR_MASK,
+    IPAddressConflict = 34 | ERROR_MASK,
+    HTTPError = 35 | ERROR_MASK,
+};
+
+// EFI_HANDLE, A collection of related interfaces
+using Handle = FlatPtr;
+
+// EFI_EVENT, Handle to an event structure
+using Event = void*;
+
+// EFI_TPL, Task priority level
+using TPL = FlatPtr;
+
+// EFI_TABLE_HEADER, Data structure that precedes all of the standard EFI table types.
+// https://uefi.org/specs/UEFI/2.10/04_EFI_System_Table.html#id4
+struct TableHeader {
+    u64 signature;
+    u32 revision;
+    u32 header_size;
+    u32 crc32;
+    u32 reserved;
+};
+static_assert(AssertSize<TableHeader, 24>());
+
+// EFI_TIME: https://uefi.org/specs/UEFI/2.10/08_Services_Runtime_Services.html#gettime
+struct Time {
+    u16 year;  // 1900 - 9999
+    u8 month;  // 1 - 12
+    u8 day;    // 1 - 31
+    u8 hour;   // 0 - 23
+    u8 minute; // 0 - 59
+    u8 second; // 0 - 59
+    u8 pad1;
+    u32 nanosecond; // 0 - 999,999,999
+    i16 time_zone;  // —1440 to 1440 or 2047
+    u8 daylight;
+    u8 pad2;
+};
+
+// EFI_PHYSICAL_ADDRESS: See "Related Definitions" at https://uefi.org/specs/UEFI/2.10/07_Services_Boot_Services.html#efi-boot-services-allocatepages
+using PhysicalAddress = u64;
+
+// EFI_VIRTUAL_ADDRESS: See "Related Definitions" at https://uefi.org/specs/UEFI/2.10/07_Services_Boot_Services.html#efi-boot-services-getmemorymap
+using VirtualAddress = u64;
+
+}
+
+template<>
+struct AK::Formatter<Kernel::EFI::Status> : Formatter<StringView> {
+    ErrorOr<void> format(FormatBuilder& builder, Kernel::EFI::Status status)
+    {
+        using enum Kernel::EFI::Status;
+        switch (status) {
+        case Success:
```

Comment:
nit: Having an `Optional<StringView> status_description(EFI::Status);` out-of-line and calling that here would be a bit nicer imho.